File: src-tauri/src/models/equipment.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use uuid::Uuid;

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct EquipmentList {
    pub telescopes: HashMap<Uuid, Telescope>,
    pub cameras: HashMap<Uuid, Camera>,
    pub mounts: HashMap<Uuid, Mount>,
    pub filters: HashMap<Uuid, Filter>,
    pub flatteners: HashMap<Uuid, Flattener>,
}

impl EquipmentList {
    pub fn new() -> Self {
        EquipmentList {
            telescopes: HashMap::new(),
            cameras: HashMap::new(),
            mounts: HashMap::new(),
            filters: HashMap::new(),
            flatteners: HashMap::new(),
        }
    }
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Telescope {
    pub id: Uuid,
    pub brand: String,
    pub name: String,
    pub focal_length: i32,
    pub aperture: i32,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Camera {
    pub id: Uuid,
    pub brand: String,
    pub name: String,
    pub pixel_size: f64,
    pub pixel_x: u32,
    pub pixel_y: u32,
    pub is_monochrome: bool,
    pub is_dslr: bool,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Mount {
    pub id: Uuid,
    pub brand: String,
    pub name: String,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Filter {
    pub id: Uuid,
    pub brand: String,
    pub name: String,
    pub filter_type: String,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Flattener {
    pub id: Uuid,
    pub brand: String,
    pub name: String,
    pub factor: f64,
}

File: src/types/equipment.ts
import { UUID } from 'crypto'

// typescript mirror of the rust equipment models in src-tauri/src/models/equipment.rs
// keys are snake_case to match the serde json output (the rust structs use no rename)
// Uuid serializes to a string, HashMap<Uuid, T> to a Record<UUID, T>

export interface Telescope {
  id: UUID
  brand: string
  name: string
  telescope_type: string
  focal_length: number
  aperture: number
}

export interface Camera {
  id: UUID
  brand: string
  name: string
  pixel_size: number
  pixel_x: number
  pixel_y: number
  sensor_type: string
}

export interface Mount {
  id: UUID
  brand: string
  name: string
}

export interface Filter {
  id: UUID
  brand: string
  name: string
  filter_type: string
  size: string
}

export interface Flattener {
  id: UUID
  brand: string
  name: string
  flattener_type: string
  factor: number
}

export interface EquipmentList {
  telescopes: Record<UUID, Telescope>
  cameras: Record<UUID, Camera>
  mounts: Record<UUID, Mount>
  filters: Record<UUID, Filter>
  flatteners: Record<UUID, Flattener>
}
